lm_eval/tasks/__init__.py

```diff
@@ -150,13 +151,16 @@ def _load_task(config, task):
                     **config,
                 }
             if self._config_is_python_task(config):
-                task_object = config["class"](config=config)
+                task_object = (
+                    config["class"](config=config)
+                    if isinstance(config["class"], ConfigurableTask)
+                    else config["class"]()
+                )
+                # very scuffed: set task name here TODO: fixme?
+                task_object.config.task = config["task"]
             else:
                 task_object = ConfigurableTask(config=config)
```

Comment:
The original implementation I think already does the what line 154-160 here intends to do.
```
            if self._config_is_python_task(config):
                task_object = config["class"](config=config)
            else:
                task_object = ConfigurableTask(config=config)
```

Reply:
hm, so this was broken when I tested it--specifically, if a Python task subclasses `Task` but not `ConfigurableTask` then things break. I think that I've now caught every check where we check `isinstance(x, ConfigurableTask)` but actually want to do `isinstance(x, Task)` instead.

Reply:
Also, sometimes people subclass `ConfigurableTask` for their Python tasks but don't implement the ability to take in a config at init, which more so falls under the scope of incorrect behavior

Reply:
Hmm, I see. In that case yeah let's add this additional check.